ios/CodePush/CodePush.m

```diff
@@ -61,7 +62,11 @@ @implementation CodePush {
 
 + (NSURL *)binaryBundleURL
 {
-    return [[NSBundle mainBundle] URLForResource:bundleResourceName withExtension:bundleResourceExtension];
+    NSBundle *bundle = [NSBundle mainBundle];
+    if (isAppExtension && [[bundle.bundleURL pathExtension] isEqualToString:@"appex"]) {
```

Comment:
Do we actually need to maintain the `isAppExtension` variable? It seems like a generally accepted solution to look for the `apex extension in order to detect whether the current executable is an app or an app extension. I'd love to not need to create the three additional `bundleURLForAppExtension` methods unless we absolutely need them.